supervisor/docker/network.py

```
@@ -42,55 +68,59 @@ def containers(self) -> list[str]:
+
+    def _get_network(self, enable_ipv6: bool = False) -> docker.models.networks.Network:
         """Get supervisor network."""
         try:
-            return self.docker.networks.get(DOCKER_NETWORK)
+            if (network := self._get(DOCKER_NETWORK)).attrs.get(
+                DOCKER_ENABLEIPV6
+            ) == enable_ipv6:
+                return network
+            network.remove()
+            _LOGGER.info(
+                f"Migrating Supervisor network to {'IPv4/IPv6 Dual Stack' if enable_ipv6 else 'IPv4-Only'}"
```

Ruff (0.11.9)

Logging statement uses f-string

(G004)